### PR TITLE
Use system assigned port instead of random during bootstrap

### PR DESCRIPTION
**Pull request by tysmith**

### CODE REVIEW

File: src/ffpuppet/bootstrapper.py [hunk 1]
```diff
@@ -18,39 +16,56 @@
 
 
 class Bootstrapper:  # pylint: disable=missing-docstring
-    BUF_SIZE = 4096  # receive buffer size
-    POLL_WAIT: float = 1  # duration of initial blocking socket operations
-    PORT_MAX = 0xFFFF  # bootstrap range
-    PORT_MIN = 0x3000  # bootstrap range
```
Comment (jschwartzentruber) [inline src/ffpuppet/bootstrapper.py:24]: Why not keep `PORT_MIN` instead of copying `BLOCKED_PORTS`?

File: src/ffpuppet/bootstrapper.py [hunk 2]
```diff
@@ -18,39 +16,56 @@
...
-    PORT_RETRIES = 100  # number of attempts to find an available port
```
Comment (jschwartzentruber) [inline src/ffpuppet/bootstrapper.py:25]: You could have kept this instead of hardcoding 50 in the `__init__` args.

**Review by jschwartzentruber — APPROVED** [review]

File: src/ffpuppet/bootstrapper.py (hunk 1 above)
Comment (tysmith) [reply src/ffpuppet/bootstrapper.py:24]: Using the minimum blocks way more ports than needed, 11264 if we use 0x3000 only 22 are blocked.

File: src/ffpuppet/bootstrapper.py (hunk 2 above)
Comment (tysmith) [reply src/ffpuppet/bootstrapper.py:25]: Agreed.

**Pull request merged**